activesupport/lib/active_support/core_ext/object/with.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class Object
+  # Sets and restore the provided attributes around a block.
+  #
+  #   client.timeout # => 5
+  #   client.with(timeout: 1) do
+  #     client.timeout # => 1
+  #   end
+  #   client.timeout # => 5
+  #
+  # This method is a shorthand for the common begin/ensure pattern:
+  #
+  #   old_value = object.attribute
+  #   begin
+  #     object.attribute = new_value
+  #     # do things
+  #   ensure
+  #     object.attribute = old_value
+  #   end
+  #
+  # It can be used on any object as long as both the reader and writer methods
+  # are public.
+  def with(**attributes)
```

Comment:
Tthis method name is super generic. Very easy to create confusion. Maybe call it `#with_attributes` instead of just a generic `#with`?